1-raw-data/mailinglists/googlegroups/googlegroups_data.go

```
@@ -0,0 +1,429 @@
+
+	// Convert string based on 1 or 2 digit month or day
+	switch numDigMonthDay {
+	case "11":
+		if tempDate, err = time.Parse("1/2/06", matchDate); err != nil {
+			err = fmt.Errorf("End date string conversion to DateTime threw an error: %v", err)
+		}
+	case "12":
+		if tempDate, err = time.Parse("1/02/06", matchDate); err != nil {
+			err = fmt.Errorf("End date string conversion to DateTime threw an error: %v", err)
+		}
+	case "21":
+		if tempDate, err = time.Parse("01/2/06", matchDate); err != nil {
+			err = fmt.Errorf("End date string conversion to DateTime threw an error: %v", err)
+		}
+	case "22":
+		if tempDate, err = time.Parse("01/02/06", matchDate); err != nil {
+			err = fmt.Errorf("End date string conversion to DateTime threw an error: %v", err)
+		}
+	}
+	// Found error in time.Parse of 2 date year that applies 20 to anything below 69. At time of thos code, anything after 2020 is future
+	if tempDate.Year() > time.Now().Year() {
+		tempDate = tempDate.AddDate(-100, 0, 0)
+	}
+
+	fileName = fmt.Sprintf("%04d-%02d.txt", tempDate.Year(), int(tempDate.Month()))
+	return
+}
+
+// Get the total topics posted at the top of the list to track all are pulled
+func getTotalTopics(dom *goquery.Document) (totalTopics int, err error) {
+	regTotal, _ := regexp.Compile("[^<]*?([0-9]+) *- *([0-9]+) of ([0-9]+)[^<]*?")
+
+	text := dom.Find("i").Text()
+
+	if regTotal.MatchString(text) {
+		match := regTotal.FindStringSubmatch(text)
+		num1, _ := strconv.Atoi(match[1])
+		num2, _ := strconv.Atoi(match[2])
+		num3, _ := strconv.Atoi(match[3])
+		numList := []int{num1, num2, num3}
+		sort.Ints(numList)
+		totalTopics = numList[2]
+	}
+	return
+}
+
+// Get message ids from topic pages and create list of raw msg urls by month
+func getMsgIDsFromDom(org, topicId, groupName string, dom *goquery.Document) (rawMsgUrl string, err error) {
+	regTopicURL, _ := regexp.Compile(fmt.Sprintf("/d/msg/%s", groupName))
```

If you're ignoring the error here, consider using the MustCompile variant of this call (and look for it elsewhere in the stdlib).

The idea behind these functions is that they in-line their error checking and panic if there is an error.